app/javascript/components/admin/UserModal.js

```diff
@@ -0,0 +1,145 @@
+      jurisdictionPath: this.props.initialUserData.jurisdiction_path ? this.props.initialUserData.jurisdiction_path : this.props.jurisdictionPaths[0],
+      role: this.props.initialUserData.role ? this.props.initialUserData.role : this.props.roles[0],
+      isAPIEnabled: this.props.initialUserData.is_api_enabled ? this.props.initialUserData.is_api_enabled : false,
+      isLocked: this.props.initialUserData.is_locked ? this.props.initialUserData.is_locked : false,
+    };
+  }
+
+  handleEmailChange = e => {
+    const val = e.target.value;
+    this.setState({ email: val });
+  };
+
+  handleJurisdictionChange = e => {
+    const val = e.target.value;
+    this.setState({ jurisdictionPath: val });
+  };
+
+  handleRoleChange = e => {
+    const val = e.target.value;
+    this.setState({ role: val });
+  };
+
+  handleLockedStatusChange = event => {
+    const val = event.target.checked;
+    this.setState({ isLocked: val });
+  };
+
+  handleAPIAccessChange = event => {
+    const val = event.target.checked;
+    this.setState({ isAPIEnabled: val });
+  };
+
+  render() {
+    return (
+      <Modal show={this.props.show} onHide={this.props.onClose} backdrop="static" aria-labelledby="contained-modal-title-vcenter" centered>
+        <Modal.Header closeButton>
+          <Modal.Title>{this.props.title}</Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <Form>
+            <Form.Group>
+              <Form.Label>Email Address</Form.Label>
+              <InputGroup>
+                <InputGroup.Prepend>
+                  <InputGroup.Text id="email-addon">
+                    <i className="fas fa-envelope"></i>
+                  </InputGroup.Text>
+                </InputGroup.Prepend>
+                <FormControl
+                  id="email-input"
+                  name="email"
+                  defaultValue={this.props.initialUserData.email ? this.props.initialUserData.email : ''}
+                  placeholder="Enter email address"
+                  aria-label="Enter email address"
+                  aria-describedby="email-addon"
+                  onChange={this.handleEmailChange}
+                />
+              </InputGroup>
+            </Form.Group>
+            <Form.Group>
+              <Form.Label>Jurisdiction</Form.Label>
+              <Form.Control
+                id="jurisdiction-input"
+                name="jurisdiction"
+                as="select"
+                onClick={this.handleJurisdictionChange}
+                defaultValue={this.props.initialUserData.jurisdiction_path ? this.props.initialUserData.jurisdiction_path : this.props.jurisdictionPaths[0]}>
+                {this.props.jurisdictionPaths.map((path, index) => {
+                  return <option key={index}>{path}</option>;
+                })}
+              </Form.Control>
+            </Form.Group>
+          </Form>
+          <Form.Group>
+            <Form.Label>Role</Form.Label>
+            <Form.Control
```

Comment:
Given the large number of jurisdictions, I wonder if this should be a datalist like it is in the change jurisdiction control on the patient view

Reply:
agreed

Reply:
Oh for the option of searching? Yeah that makes sense. I'll make that change.

Reply:
Just getting to this. The autocomplete is turned off in the change jurisdiction control in the patient view. What exactly is the reason for switching to datalist?

Reply:
Autocomplete. I'm not a frontend expert -- maybe there's another way to accomplish selecting an existing value without needing to scroll through hundreds of values but my mind immediately went to datalist since that's what ive seen used to accomplish this before.

Reply:
Yeah that makes sense. Just wanted to clarify since we aren't doing autocomplete for the datalist for change jurisdiction on the patient view.

Reply:
Datalist has the unfortunate consequence of needing to be empty to show the whole dropdown. I just remembered Sarah brought in react-select to address this problem exactly for the language dropdowns, so I'm going to use that.

Reply:
ooh good idea! we might be able to replace all the other jurisdiction inputs in the future with this as well

Reply:
Done.